### DISCUSSION TITLE
error "One of the identified items was in an invalid format." for .Items.GetById(125)

### DISCUSSION THREAD

**Discussion by Ofer-Gal** [pnp/pnpcore#753] [Q&A]
When the file I try to get with `.Items.GetById(125)` came from another site in SPO, I get error:
"One of the identified items was in an invalid format."
If I manually publish the document, then try again, it works fine.

What could be the issue and is there a different syntax I could use to overcome the issue?
Thank You

**Comment by jansenbe** [comment]
@Ofer-Gal : would you be able to provide more error details like stack trace or Fiddler trace?

**Reply by Ofer-Gal** [reply]
Now I can't get it to error :-( 
I also wanted to supply the optional parameters besides the item ID. 
I saw in the documentations something about versions and fields. but I could not figure out how to supply them.
Could you help with code that will ask for any version (when I had the error, I only had minor versions, never published ) also I want to bring over these fields:
Requester Person
DocumentNumber string
ProjectNumber string
RequestNumber string
DocumentType taxonomy

Appreciate the help

**Reply by jansenbe** [reply]
@Ofer-Gal : for list item versions check this: https://pnp.github.io/pnpcore/using-the-sdk/listitems-intro.html#getting-list-item-versions, for field handling check this: https://pnp.github.io/pnpcore/using-the-sdk/listitems-fields.html

**Reply by Ofer-Gal** [reply]
I was referring to [GetById](https://pnp.github.io/pnpcore/api/PnP.Core.Model.SharePoint.IListItemVersionCollection.html#PnP_Core_Model_SharePoint_IListItemVersionCollection_GetById_System_Int32_Expression_Func_PnP_Core_Model_SharePoint_IListItemVersion_System_Object_____) and [GetByIdAsync](https://pnp.github.io/pnpcore/api/PnP.Core.Model.SharePoint.IListItemVersionCollection.html#PnP_Core_Model_SharePoint_IListItemVersionCollection_GetByIdAsync_System_Int32_Expression_Func_PnP_Core_Model_SharePoint_IListItemVersion_System_Object_____)  
How do I build the "selectors" with my fields? Maybe that will prevent the error.

**Reply by jansenbe** [reply]
@Ofer-Gal : the custom fields should be loaded automatically, if you want additional list item properties you can specify them in the selector. E.g. ´await myList.Items.GetByIdAsync(125, li => li.All, li => li.Versions);` loads the list item with all standard REST properties + loads the list item versions

**Reply by Ofer-Gal** [reply]
Thank you
